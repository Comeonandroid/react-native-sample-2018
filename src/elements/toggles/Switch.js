import React from 'react'
import { MKSwitch } from 'react-native-material-kit'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

const Switch = ({ children, checked, ...rest }) => (
  <Wrapper>
    <Input
      checked={checked}
      onColor='#fdc77c'
      thumbOnColor='#faa632'
      rippleColor={`rgba(250,166,50,.15)`}
      {...rest}
    />
    <Label>{children}</Label>
  </Wrapper>
)

const Wrapper = styled.View `
  flex-direction: column;
  align-items: center;
  width: auto;
  align-self: flex-start;
`

const Input = styled(MKSwitch) `
`

const Label = styled.Text `
`

Switch.propTypes = {
  children: PropTypes.node,
  checked: PropTypes.bool
}

Switch.defaultProps = {
  checked: false
}

export default Switch