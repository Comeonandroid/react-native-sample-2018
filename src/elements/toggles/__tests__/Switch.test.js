import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'

import { itBehavesLike } from '../../../specs/shared_examples/index'

import Switch from '../Switch'

const setup = propOverrides => {
  const props = Object.assign({
    children: null,
    checked: false,
    onCheckedChange: jest.fn(),
  }, propOverrides)

  const component = <Switch {...props} />
  const wrapper = shallow(component)

  return {
    props,
    wrapper,
    component
  }
}

describe('<Switch />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  it('renders correctly with children', () => {
    const { props, wrapper } = setup({ children: 'Test' })

    expect(wrapper.children().children().last().text()).toEqual(props.children)
  })
})
