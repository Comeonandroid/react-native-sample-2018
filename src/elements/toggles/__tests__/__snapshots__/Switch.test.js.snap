// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flexDirection": "column",
        "width": "auto",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
    style={
      Array [
        Object {},
        undefined,
      ]
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.25)",
          "borderRadius": 10,
          "height": 20,
          "margin": 18,
          "width": 48,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "left": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "transparent",
            "height": 56,
            "position": "absolute",
            "width": 56,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "elevation": 2,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.7,
              "shadowRadius": 1,
            },
            Object {
              "borderRadius": 14,
              "height": 28,
              "left": 14,
              "top": 14,
              "width": 28,
            },
            Object {
              "backgroundColor": "#EAEAEA",
            },
          ]
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "rgba(250,166,50,.15)",
            "borderRadius": 28,
            "height": 56,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
            "width": 56,
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  />
</View>
`;

exports[`<Switch /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flexDirection": "column",
        "width": "auto",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
    style={
      Array [
        Object {},
        undefined,
      ]
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.25)",
          "borderRadius": 10,
          "height": 20,
          "margin": 18,
          "width": 48,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "left": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "transparent",
            "height": 56,
            "position": "absolute",
            "width": 56,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "elevation": 2,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.7,
              "shadowRadius": 1,
            },
            Object {
              "borderRadius": 14,
              "height": 28,
              "left": 14,
              "top": 14,
              "width": 28,
            },
            Object {
              "backgroundColor": "#EAEAEA",
            },
          ]
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "rgba(250,166,50,.15)",
            "borderRadius": 28,
            "height": 56,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
            "width": 56,
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  />
</View>
`;
